app/models/response/ResponseMessage.scala

```diff
@@ -18,17 +18,18 @@ package models.response
 
 import java.time.LocalDateTime
 
-import controllers.routes
 import models.domain.MovementMessage
 import play.api.libs.json.Json
+import play.api.mvc.Call
 import utils.NodeSeqFormat
 import utils.CallOps._
+
 import scala.xml.NodeSeq
 
 object ResponseMessage extends NodeSeqFormat{
   implicit val format = Json.format[ResponseMessage]
 
-  def apply(m: MovementMessage, arrivalId: String, messageId: String): ResponseMessage = ResponseMessage(routes.ArrivalMessagesController.getArrivalMessage(arrivalId, messageId).urlWithContext, m.dateTime, m.messageType, m.message)
+  def apply(m: MovementMessage, call: Call): ResponseMessage = ResponseMessage(call.urlWithContext, m.dateTime, m.messageType, m.message)
```

Comment:
Can .urlWithContext be passed into MovementMessage?

I think the apply method should take in a single MovementMessage argument, with location being passed in to the MovementMessage constructor.

Reply:
1) I considered that, but this way we separate the presentation of the call from the call itself.
2) I can't do anything about using a single argument, since the movementmessage is the deserialized object that I'm mapping from that belongs to one of two backend apis

Reply:
:ok_hand: